mitodl/mit-open
# Capture page views with more information

### What are the relevant tickets?

Closes mitodl/hq#3926

### Description (What does it do?)

Adds more descriptive page view events to the Learning Resource Drawer. 

The base PostHogProvider code that was added earlier captures a basic page view event which includes a set of PostHog data and the page URL. For the learning resource drawer, we'd like to track a bit more data to make it easier to identify views for learning resources and to make them easier to aggregate. 

This adds a component that gets rendered alongside the drawer content when it's open and fires a custom `lrd_view` with the following information:
- `resourceId`: the database ID of the resource
- `readableId`: the readable ID of the resource (courses-v1:MITxT+etc, etc.)
- `platformCode`: the code for the platform that the resource belongs to
- `resourceType`: the type of resource (course, podcast, etc.) 

In addition, this fixes a small bug whereby disabling the PostHog integration in the environment would not actually do that - you had to remove the `POSTHOG_ENABLED` key (since env values are strings, so "False" was also truthy). Now it will check for `POSTHOG_ENABLED=True` specifically; any other value (or capitalization) will be regarded as JS `false`.

### How can this be tested?

Automated tests should pass.

You will need a PostHog instance or account to test this. Once you have that set up, you should be able to see `lrd_view` events come in. Clicking on the arrows next to the event name should show you the additional information captured with the event:
![image](https://github.com/mitodl/mit-open/assets/945611/68b06d8d-2a90-46fa-b480-4be1da5d5c67)

If PostHog is disabled, this should do nothing. 

### Additional Context

Next step is getting this out of PostHog and into OpenSearch
